contracts/ronin/validator/CandidateManager.sol

```diff
@@ -135,17 +139,25 @@ abstract contract CandidateManager is ICandidateManager, PercentageConsumer, Glo
   /**
    * @inheritdoc ICandidateManager
    */
-  function isValidatorCandidate(address _addr) public view override returns (bool) {
-    return _candidateIndex[_addr] != 0;
+  function isValidatorCandidate(address _addr) public view override returns (bool yes) {
+    assembly {
+      mstore(0x00, _addr)
+      mstore(0x20, _candidateIndex.slot)
+      yes := iszero(iszero(sload(keccak256(0x00, 0x40))))
```

Comment:
I think this line is quite overkill.

Reply:
propose modification is mainly for guiding the compiler to follow this optimization rule: https://github.com/ethereum/solidity/issues/9923